concept/thing/thing_manager.rs

```diff
@@ -866,23 +1093,59 @@ impl ThingManager {
             }
         }
 
+        let mut any_deleted = true;
```

Comment:
The following code is related to relations and attributes cleanups. 
I could optimize it to iterate through all writes just once instead of copying the buffer every time just like we did (with an additional `clone` =) ) in the locks function, but it's easier to scale the code this way for now.

Reply:
I could also try to write general functions to reduce code volumes, but I didn't do it for the same reason...